### ISSUE TITLE
Usage Example

### ISSUE THREAD

**Issue by nickrobillard** [thegallagher/PHP-MP3#1]
So I might be missing something obvious but I cannot get the extract() method to work. It always saves the entire file no matter what combination of start and length I use.

Here is my code:

``` php
require_once 'phpmp3.php';
$mp3 = new PHPMP3('path-to-input-file.mp3');
$mp3->extract(20, 10);
$mp3->save('path-to-output-file.mp3');
```

I expect this to return 10 seconds of the file beginning at the 20 second mark but it simply saves the entire MP3 file.

It'd be really cool for you to include a small example in the README. :) Appreciate any feedback you have on this.

**Comment by nickrobillard** [comment]
Ah I found my problem.

``` php
$mp3->extract(20, 10);
```

needed to be

``` php
$mp3 = $mp3->extract(20, 10);
```

A usage example would help others avoid this kind of mistake in the future. :) Thanks for hosting this script!

**Comment by nickrobillard** [comment]
Another usage question. Do you have an example that extracts from a specific start time to the very end of the MP3? For example, similar classes allow one to pass in -1 as the length or end argument, which signifies the end of file.

**Comment by thegallagher** [comment]
It's been a long time since I have used this class but as far as I can tell passing -1 as length will allow you to extract from a specific time to the end of the MP3. Let me know if that works.

This project could certainly use some documentation and a bit of love too. There are many parts of the class which could be made better and some documentation would help a lot. As this project is starting to receive a bit more attention lately, I would like to put a bit more work into it when I find the time.

I hope you do get it all working and if you have some time yourself, feel free to submit a pull request with documentation or improvements.

**Comment by nickrobillard** [comment]
Thanks for your response. I've used -1 and it does not work as I expect. Depending on start time, it sometimes returns a few seconds, other times many seconds, but not the rest of the file. I haven't gone down that debug road yet. Was hoping I was missing something simple.

Fyi, the reason I'm interested in this script is because I'm attempting to get MP3 pseudo streaming working on dubplate.fm without any goofie workarounds that require muxing an FLV with the MP3, etc. etc. (second post here: http://www.longtailvideo.com/support/forums/jw-player/setup-issues-and-embedding/7606/fast-skippingpseudo-streaming-mp3s)

On a side note, I have a feeling doing the processing in PHP might get too resource intensive. If so, I will have to abandon this script for a Python or FFMPEG command line alternative (probably the latter). Do you have any data on the performance of this class? I'm dealing with DJ sets up to 200 MB in size and I'm already noticing some 2 - 8 second wait times for the new file to be written. I imagine 20 users concurrently seeking through big sets might just be too much. This class is pretty sweet nonetheless! So far, it's the only one that doesn't choke on my MP3s (which are user uploaded and have minor encoding differences).

**Comment by thegallagher** [comment]
It would be awesome if you did get that working with pure PHP but you would need to do some optimisations to this class for it to work smoothly, especially with files the size you mentioned.

The class loads the full MP3 into memory which especially in your case, will use a ton of memory. I have noted before that the class should not do this and should use `fopen` and friends to read and write the files. If you did this, the class would be use a much less memory.

In your case I assume you can output the MP3 data directly to the client. If the extract method just calculated the start and end frames then you could also create a read method for the class to send the MP3 data piece by piece. This should allow you to handle quite a few users.

Still if you plan on having a lot of users, it would make more sense to use something tried and tested.

**Comment by thegallagher** [comment]
There is another class here http://www.phpclasses.org/browse/file/26606.html which might be a little faster. I haven't tried it though.

**Comment by nickrobillard** [comment]
Yeah I noticed your comments about fseek support. Sounds like that would definitely help things.

And yes, I tried class.mp3.php but it choked on most of my mp3s. It failed at returning the frames for all except 1 MP3 that I tested with (and didn't debug much further). The MP3 that worked was a small file, so I didn't notice any real performance difference.

Honestly, I think my best bet is to use ffmpeg from command line using proc_open() or passthru(). Ala, first answer here http://stackoverflow.com/questions/43890/crop-mp3-to-first-30-seconds. All of the disk writing required is probably better handled by something lower level.
